Code/Framework/AzFramework/AzFramework/IO/LocalFileIO.cpp

```diff
@@ -707,17 +707,21 @@ namespace AZ
                 resolvedPathLen += postAliasView.size();
                 // Null-Terminated the resolved path
                 resolvedPath[resolvedPathLen] = '\0';
+
                 // If the path started with one of the "asset cache" path aliases, lowercase the path
                 const char* assetAliasPath = GetAlias("@assets@");
                 const char* rootAliasPath = GetAlias("@root@");
                 const char* projectPlatformCacheAliasPath = GetAlias("@projectplatformcache@");
                 const bool lowercasePath = (assetAliasPath != nullptr && AZ::StringFunc::StartsWith(resolvedPath, assetAliasPath)) ||
                     (rootAliasPath != nullptr && AZ::StringFunc::StartsWith(resolvedPath, rootAliasPath)) ||
                     (projectPlatformCacheAliasPath != nullptr && AZ::StringFunc::StartsWith(resolvedPath, projectPlatformCacheAliasPath));
+
                 if (lowercasePath)
                 {
-                    AZStd::to_lower(resolvedPath, resolvedPath + resolvedPathLen);
+                    // Lowercase only the relative part after the replaced alias.
+                    AZStd::to_lower(resolvedPath + aliasValue.size(), resolvedPath + resolvedPathLen);
```

Comment:
Is it possible for aliasValue.size() > resolvedPathLen?

Reply:
there are checks on 688-694 for that

Reply:
At this point, `resolvedPathLen` should be at least as big as `aliasValue.size()`.  The `resolvedPath` buffer contains the `aliasValue` and the remaining part of the path, and `resolvedPathLen` was updated each time those parts were added.